cli/core/pkg/interfaces/config_client_interface.go

```diff
@@ -22,3 +24,13 @@ func NewConfigClient() ConfigClient {
 func (cc *configClientImpl) GetEnvConfigurations() map[string]string {
 	return config.GetEnvConfigurations()
 }
+
+func (cc *configClientImpl) StoreClientConfig(clientConfig *configapi.ClientConfig) error {
+	// Acquire tanzu config lock
+	config.AcquireTanzuConfigLock()
+	defer config.ReleaseTanzuConfigLock()
```

Comment:
Adding this means `configClientImpl.StoreClientConfig ` is not same as actual `config.StoreClientConfig` which is concerning to me. Why do we need update the logic? Can we keep it just as a wrapper and not add things like this?

If needed we could add additional functions like `AcquireTanzuConfigLock` and `ReleaseTanzuConfigLock` to `configClientImpl` but this should only be treated as wrapper.

Reply:
Curious, getting the lock and releasing the lock shouldn't be part of function `config.StoreClientConfig(clientConfig)`? whats the reason we are getting and releasing the lock outside of it?